pkg/task/v1alpha1/runtask_command.go

```diff
@@ -0,0 +1,328 @@
+	Category    RunCommandCategoryList  // classification of runtask command
+	Data        RunCommandDataMap       // input data required to execute runtask command
+	SelectPaths []string                // paths whose values will be retrieved after runtask command execution
+	*Msgs                               // store and retrieve info, warns, errors, etc occured during execution
+}
+
+// Command returns a new instance of RunCommand
+func Command() *RunCommand {
+	return &RunCommand{Msgs: &Msgs{}, WillRun: true}
+}
+
+// AddError updates RunCommand instance with given error
+func (c *RunCommand) AddError(err error) (u *RunCommand) {
+	c.Msgs.AddError(err)
+	return c
+}
+
+// CreateAction updates RunCommand instance with create action
+func (c *RunCommand) CreateAction() (u *RunCommand) {
+	c.Action = CreateCommandAction
+	return c
+}
+
+// DeleteAction updates RunCommand instance with delete action
+func (c *RunCommand) DeleteAction() (u *RunCommand) {
+	c.Action = DeleteCommandAction
+	return c
+}
+
+// GetAction updates RunCommand instance with get action
+func (c *RunCommand) GetAction() (u *RunCommand) {
+	c.Action = GetCommandAction
+	return c
+}
+
+// ListAction updates RunCommand instance with list action
+func (c *RunCommand) ListAction() (u *RunCommand) {
+	c.Action = ListCommandAction
+	return c
+}
+
+// UpdateAction updates RunCommand instance with update action
+func (c *RunCommand) UpdateAction() (u *RunCommand) {
+	c.Action = UpdateCommandAction
+	return c
+}
+
+// PatchAction updates RunCommand instance with patch action
+func (c *RunCommand) PatchAction() (u *RunCommand) {
+	c.Action = PatchCommandAction
+	return c
+}
+
+func (c *RunCommand) String() string {
+	return fmt.Sprintf("run command: id='%s', action='%s', willrun='%t', category='%s', data='%s', select='%s'", c.ID, c.Action, c.WillRun, c.Category, c.Data, strings.Join(c.SelectPaths, " "))
+}
+
+func (c *RunCommand) Result(result interface{}) (r RunCommandResultMap) {
+	return NewRunCommandResultMap(result, c.All())
+}
+
+// instance fetches the specific runtask command implementation instance based
+// on command categories
+func (c *RunCommand) instance() (r CommandRunner) {
+	if c.Category.IsJivaVolume() {
+		r = &jivaVolumeCommand{cmd: c}
+	} else {
+		r = &notSupportedCategoryCommand{cmd: c}
+	}
+	return
+}
+
+// preRun does some bookkeeping stuff prior to executing the runtask command
+func (c *RunCommand) preRun() {
+	c.AddInfo(c.String())
+}
+
+// Run finds the specific runtask command implementation and executes the same
+func (c *RunCommand) Run() (r RunCommandResultMap) {
+	c.preRun()
+	return c.instance().Run()
+}
+
+// WithCategory updates the given RunCommand instance with provided category
+func WithCategory(given *RunCommand, category RunCommandCategory) (updated *RunCommand) {
+	given.Category = append(given.Category, category)
+	return given
+}
+
+// WithData updates the given RunCommand instance with provided category
```

Comment:
Typo in comment "with provided category" should be "with provided input data"

Reply:
will address in next commit